x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/Chunk.java

```diff
@@ -0,0 +1,240 @@
+import org.elasticsearch.xpack.esql.core.expression.Literal;
+import org.elasticsearch.xpack.esql.core.expression.TypeResolutions;
+import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.expression.function.Example;
+import org.elasticsearch.xpack.esql.expression.function.FunctionInfo;
+import org.elasticsearch.xpack.esql.expression.function.Param;
+import org.elasticsearch.xpack.esql.expression.function.TwoOptionalArguments;
+import org.elasticsearch.xpack.esql.expression.function.scalar.EsqlScalarFunction;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamInput;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.ParamOrdinal.FIRST;
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.ParamOrdinal.SECOND;
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.ParamOrdinal.THIRD;
+import static org.elasticsearch.xpack.esql.core.expression.TypeResolutions.isString;
+import static org.elasticsearch.xpack.esql.core.type.DataType.INTEGER;
+
+public class Chunk extends EsqlScalarFunction implements TwoOptionalArguments {
+
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(Expression.class, "Chunk", Chunk::new);
+
+    public static final int DEFAULT_NUM_CHUNKS = Integer.MAX_VALUE;
+    public static final int DEFAULT_CHUNK_SIZE = 300;
+
+    private final Expression field, numChunks, chunkSize;
+
+    @FunctionInfo(returnType = "keyword", preview = true, description = """
+        Use `CHUNK` to split a text field into smaller chunks.""", detailedDescription = """
+            Chunk can be used on fields from the text famiy like <<text, text>> and <<semantic-text, semantic_text>>.
+            Chunk will split a text field into smaller chunks, using a sentence-based chunking strategy.
+            The number of chunks returned, and the length of the sentences used to create the chunks can be specified.
+        """, examples = { @Example(file = "chunk", tag = "chunk-with-field", applies_to = "stack: preview 9.3.0") })
+    public Chunk(
+        Source source,
+        @Param(name = "field", type = { "keyword", "text" }, description = "The input to chunk.") Expression field,
+        @Param(
+            optional = true,
+            name = "num_chunks",
+            type = { "integer" },
+            description = "The number of chunks to return. Defaults to return all chunks."
+        ) Expression numChunks,
+        @Param(
+            optional = true,
+            name = "chunk_size",
+            type = { "integer" },
+            description = "The size of sentence-based chunks to use. Defaults to " + DEFAULT_CHUNK_SIZE
+        ) Expression chunkSize
```

Comment:
We don't want these 2 to be marked as optional.
What if we want to return all chunks, but have a non default chunk size?
Instead of them be positional params, we should consider using a `MapExpression` here (like we do for match_phrase options for example).
So can support something like `chunk(content, {"chunk_size": 25 })`.